docs/source/user_guide/installation/windows-wsl.rst

```diff
@@ -1,108 +1,40 @@
-Windows Subsystem for Linux (WSL)
-======================================
+Install from source (Windows Subsystem for Linux)
+=================================
 
-We recommend the use of Windows Subsystem for Linux (WSL) to install
-PyBaMM, see the instructions below to get PyBaMM working using Windows,
-WSL and VSCode.
-
-.. contents::
+To make it easier to install PyBaMM, we recommend using the Windows Subsystem for Linux (WSL) along with Visual Studio Code. This guide will walk you through the process.
 
 Install WSL
 -----------
 
-Follow the instructions from Microsoft
-`here <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__.
-When given the option, choose the Ubuntu 18.04 LTS distribution to
-install. Best practices for setting up a WSL development environment can be found
-`here <https://docs.microsoft.com/en-us/windows/wsl/setup/environment>`__.
+Install Ubuntu 22.04 or 20.04 LTS as a distribution for WSL following `Microsoft's guide to install WSL <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`__. For a seamless development environment, refer to `this guide <https://docs.microsoft.com/en-us/windows/wsl/setup/environment>`__.
 
 Install PyBaMM
 --------------
 
-Open a terminal window in your installed Ubuntu distribution by
-selecting “Ubuntu” from the start menu. This should give you a bash
-prompt in your home directory.
+Get PyBaMM's Source Code
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Open a terminal in your Ubuntu distribution by selecting "Ubuntu" from the Start menu. You'll get a bash prompt in your home directory.
 
-To download the PyBaMM source code, you first need to install git, which
-you can do by typing
+2. Install Git by typing the following command:
 
 .. code:: bash
 
-   sudo apt install git-core
+     sudo apt install git-core
 
-For easier integration with WSL, we recommend that you install PyBaMM in
-your *Windows* Documents folder, for example by first navigating to
+3. Clone the PyBaMM repository::
 
 .. code:: bash
 
-   $ cd /mnt/c/Users/USER_NAME/Documents
+     git clone https://github.com/pybamm-team/PyBaMM.git
 
-where USER_NAME is your username. Exact path to Windows documents may
-vary. Now use git to clone the PyBaMM repository:
+4. Enter the PyBaMM Directory by running::
 
 .. code:: bash
 
-   git clone https://github.com/pybamm-team/PyBaMM.git
+     cd PyBaMM
 
-This will create a new directly called ``PyBaMM``, you can move to this
-directory in bash using the ``cd`` command:
-
-.. code:: bash
+5. Follow the Installation Steps
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
How did you get the plotting to work? Was it just running VS Code from Windows rather than Linux? Because for me it only works if I do the XServer thing, but I might do something wrong...

Reply:
> How did you get the plotting to work? Was it just running VS Code from Windows rather than Linux? Because for me it only works if I do the XServer thing, but I might do something wrong...

In my case I was able to plot with `python examples/scripts/run_simulation.py` straight without having to install Xserver. 

But with `python examples/scripts/create-model.py` I am getting some attribute errors locally i.e. 

```
Traceback (most recent call last):
  File "/mnt/d/Arjun/PyBaMM/examples/scripts/create_model.py", line 99, in <module>
    param.process_geometry(geometry)
  File "/mnt/c/Users/Arjun/Desktop/PyBaMM/pybamm/parameters/parameter_values.py", line 463, in process_geometry
    for lim, sym in spatial_limits.items():
AttributeError: 'Scalar' object has no attribute 'items'
```

Reply:
Happy to merge once this conversation is resolved.

Reply:
> In my case I was able to plot with python examples/scripts/run_simulation.py straight without having to install Xserver.

Ok, then happy to proceed with the instructions as they are.

> But with python examples/scripts/create-model.py I am getting some attribute errors locally i.e.

That's strange, as this seems to be an internal PyBaMM issue. This is not picked up by the tests so I would be tempted to say go ahead and merge this.